src/ui/window.c

```diff
@@ -656,15 +654,20 @@ win_page_up(ProfWin* window, int scroll_size)
                 win_print_loading_history(window);
                 iq_mam_request_older(chatwin);
             }
+
+            int buff_size = buffer_size(window->layout->buffer);
+            int offset_entry_id = buff_size > 10 ? 10 : buff_size - 1;
+            int offset = buffer_get_entry(window->layout->buffer, offset_entry_id)->y_end_pos;
+            *page_start = offset - page_space;
         }
     }
 
-    // went past beginning, show first page
```

Comment:
why is this comment removed? I think its helpful to quickly see whats going on here.